src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryTool.DetectBinaries.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.Extensions.Logging;
+
+namespace BinaryToolKit;
+
+public partial class BinaryTool
+{
+    private const string Utf16Marker = "UTF-16";
+    private const int ChunkSize = 4096;
+
+    private async Task<IEnumerable<string>> DetectBinariesAsync()
+    {
+        Log.LogInformation($"Detecting binaries in {TargetDirectory}...");
+
+        var matcher = new Matcher(StringComparison.Ordinal);
+        matcher.AddInclude("**/*");
+        matcher.AddExcludePatterns(new[] { "**/.dotnet/**", "**/.git/**", "**/git-info/**", "**/artifacts/**", "**/prereqs/packages/**", "**/.packages/**" });
```

Comment:
Nit: this isn't written anywhere, but staying under 120 chars per line makes code easier to read, especially for arrays like this. Also will a [collection expression](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/collection-expressions) work here?